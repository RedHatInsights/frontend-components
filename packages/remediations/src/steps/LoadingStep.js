import React, { PureComponent } from 'react';
import propTypes from 'prop-types';

import { Skeleton, SkeletonSize } from '@redhat-cloud-services/frontend-components/components/cjs/Skeleton';
import { Stack, StackItem } from '@patternfly/react-core';

import './LoadingStep.scss';

export default class LoadingStep extends PureComponent {

    componentDidMount() {
        this.props.onValidChange(false);
    }

    componentWillUnmount () {
        this.props.onValidChange(true);
    }

    render () {
        return (
            <Stack gutter='sm' className='ins-c-remediations-loading-step'>
                <StackItem><h1><Skeleton size={ SkeletonSize.sm } /></h1></StackItem>
                <StackItem>
                    <Skeleton size={ SkeletonSize.lg } className='ins-c-remediations-loading-step-main' />
                </StackItem>
                <StackItem>
                    <Stack gutter='sm'>
                        <StackItem>
                            <Skeleton size={ SkeletonSize.sm } />
                        </StackItem>
                        <StackItem>
                            <Skeleton size={ SkeletonSize.xs } />
                        </StackItem>
                    </Stack>
                </StackItem>
            </Stack>
        );
    }
}

LoadingStep.propTypes = {
    onValidChange: propTypes.func.isRequired
};

