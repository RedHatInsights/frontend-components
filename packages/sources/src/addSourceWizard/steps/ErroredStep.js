import React from 'react';
import PropTypes from 'prop-types';

import { EmptyStateBody, EmptyState, EmptyStateIcon, Title, Button, EmptyStateSecondaryActions, EmptyStateVariant } from '@patternfly/react-core';
import { ErrorCircleOIcon } from '@patternfly/react-icons';

const ErroredStep = ({ onClose, onRetry }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <EmptyStateIcon icon={ ErrorCircleOIcon } color="var(--pf-global--danger-color--100)" />
        <Title headingLevel="h5" size="lg">
      Configuration unsuccessful
        </Title>
        <EmptyStateBody>
      Something went wrong. Your source has not been successfully added.
        </EmptyStateBody>
        <Button variant="primary" onClick={ onClose }>Back to my application</Button>
        <EmptyStateSecondaryActions>
            <Button variant="link" onClick={ onRetry }>Retry</Button>
        </EmptyStateSecondaryActions>
    </EmptyState>
</div>;

ErroredStep.propTypes = {
    onClose: PropTypes.func.isRequired,
    onRetry: PropTypes.func.isRequired
};

export default ErroredStep;
