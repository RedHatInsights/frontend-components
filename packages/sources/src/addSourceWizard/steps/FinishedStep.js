import React from 'react';
import PropTypes from 'prop-types';

import { EmptyStateBody, EmptyState, EmptyStateIcon, Title, Button, EmptyStateSecondaryActions, EmptyStateVariant } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';

const FinishedStep = ({ onClose, successfulMessage }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <EmptyStateIcon icon={ CheckCircleIcon } color="var(--pf-global--success-color--100)" />
        <Title headingLevel="h5" size="lg">
      Configuration successful
        </Title>
        <EmptyStateBody>
            { successfulMessage }
        </EmptyStateBody>
        <Button variant="primary" onClick={ onClose }>Back to my application</Button>
        <EmptyStateSecondaryActions>
            <a href='/hybrid/settings/sources'>
                <Button variant="link">Take me to sources</Button>
            </a>
        </EmptyStateSecondaryActions>
    </EmptyState>
</div>;

FinishedStep.propTypes = {
    onClose: PropTypes.func.isRequired,
    successfulMessage: PropTypes.node.isRequired
};

export default FinishedStep;
