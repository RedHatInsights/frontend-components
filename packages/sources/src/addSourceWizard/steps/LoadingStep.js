import React from 'react';
import PropTypes from 'prop-types';
import { EmptyState, EmptyStateVariant, EmptyStateBody, EmptyStateSecondaryActions, Button } from '@patternfly/react-core';
import { Spinner } from '@redhat-cloud-services/frontend-components';

const LoadingStep = ({ onClose, customText }) => <div className="pf-l-bullseye">
    <EmptyState variant={ EmptyStateVariant.full }>
        <div>
            <Spinner />
        </div>
        <EmptyStateBody>
            { customText }
        </EmptyStateBody>
        { onClose &&
        <EmptyStateSecondaryActions>
            <Button variant="link" onClick={ onClose }>Cancel</Button>
        </EmptyStateSecondaryActions> }
    </EmptyState>
</div>;

LoadingStep.propTypes = {
    onClose: PropTypes.func,
    customText: PropTypes.string
};

LoadingStep.defaultProps = {
    customText: 'Loading, please wait.'
};

export default LoadingStep;
