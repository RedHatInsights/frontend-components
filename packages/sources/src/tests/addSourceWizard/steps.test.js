import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Button, EmptyStateBody } from '@patternfly/react-core';

import FinishedStep from '../../addSourceWizard/steps/FinishedStep';
import LoadingStep from '../../addSourceWizard/steps/LoadingStep';
import ErroredStep from '../../addSourceWizard/steps/ErroredStep';

describe('Steps components', () => {
    let initialProps;
    let spyFunction;
    let spyFunctionSecond;

    beforeEach(() => {
        spyFunction = jest.fn();
        spyFunctionSecond = jest.fn();
    });

    afterEach(() => {
        spyFunction.mockReset();
        spyFunctionSecond.mockReset();
    });

    describe('FinishedStep', () => {

        beforeEach(() => {
            initialProps = {
                onClose: spyFunction,
                successfulMessage: 'Here I Am',
                hideSourcesButton: false
            };
        });

        it('renders correctly', () => {
            const wrapper = shallow(<FinishedStep { ...initialProps }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('a[href="/hybrid/settings/sources"]').length).toBe(1);
            expect(wrapper.find(EmptyStateBody).html().includes('Here I Am')).toBe(true);
        });

        it('renders withou takemetosources button', () => {
            const wrapper = shallow(<FinishedStep { ...initialProps } hideSourcesButton={ true }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('a[href="/hybrid/settings/sources"]').length).toBe(0);
        });

        it('calls onClose function', () => {
            const wrapper = shallow(<FinishedStep { ...initialProps }/>);
            wrapper.find(Button).first().simulate('click');
            expect(spyFunction).toHaveBeenCalled();
        });
    });

    describe('LoadingStep', () => {

        beforeEach(() => {
            initialProps = {
                onClose: spyFunction,
                customText: 'Here I Am'
            };
        });

        it('renders correctly with custom props', () => {
            const wrapper = shallow(<LoadingStep { ...initialProps }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find(EmptyStateBody).html().includes('Here I Am')).toBe(true);
        });

        it('renders correctly', () => {
            const wrapper = shallow(<LoadingStep />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('calls onClose function', () => {
            const wrapper = shallow(<LoadingStep { ...initialProps }/>);
            wrapper.find(Button).first().simulate('click');
            expect(spyFunction).toHaveBeenCalled();
        });
    });

    describe('ErroredStep', () => {

        beforeEach(() => {
            initialProps = {
                onClose: spyFunction,
                onRetry: spyFunctionSecond
            };
        });

        it('renders correctly', () => {
            const wrapper = shallow(<ErroredStep { ...initialProps }/>);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it('calls onClose function', () => {
            const wrapper = shallow(<ErroredStep { ...initialProps }/>);
            wrapper.find(Button).first().simulate('click');
            expect(spyFunction).toHaveBeenCalled();
        });

        it('calls onRetry function', () => {
            const wrapper = shallow(<ErroredStep { ...initialProps }/>);
            wrapper.find(Button).last().simulate('click');
            expect(spyFunctionSecond).toHaveBeenCalled();
        });
    });
});
