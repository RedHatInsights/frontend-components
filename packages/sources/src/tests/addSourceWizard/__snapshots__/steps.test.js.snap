// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps components ErroredStep renders correctly 1`] = `
<div
  className="pf-l-bullseye"
>
  <EmptyState
    variant="full"
  >
    <EmptyStateIcon
      color="var(--pf-global--danger-color--100)"
      icon={[Function]}
    />
    <Title
      headingLevel="h5"
      size="lg"
    >
      Configuration unsuccessful
    </Title>
    <EmptyStateBody>
      Something went wrong. Your source has not been successfully added.
    </EmptyStateBody>
    <Button
      onClick={[MockFunction]}
      variant="primary"
    >
      Go back to my application
    </Button>
    <EmptyStateSecondaryActions>
      <Button
        onClick={[MockFunction]}
        variant="link"
      >
        Retry
      </Button>
    </EmptyStateSecondaryActions>
  </EmptyState>
</div>
`;

exports[`Steps components FinishedStep renders correctly 1`] = `
<div
  className="pf-l-bullseye"
>
  <EmptyState
    variant="full"
  >
    <EmptyStateIcon
      color="var(--pf-global--success-color--100)"
      icon={[Function]}
    />
    <Title
      headingLevel="h5"
      size="lg"
    >
      Configuration successful
    </Title>
    <EmptyStateBody>
      Here I Am
    </EmptyStateBody>
    <Button
      onClick={[MockFunction]}
      variant="primary"
    >
      Go back to my application
    </Button>
    <EmptyStateSecondaryActions>
      <a
        href="/hybrid/settings/sources"
      >
        <Button
          variant="link"
        >
          Take me to sources
        </Button>
      </a>
    </EmptyStateSecondaryActions>
  </EmptyState>
</div>
`;

exports[`Steps components FinishedStep renders withou takemetosources button 1`] = `
<div
  className="pf-l-bullseye"
>
  <EmptyState
    variant="full"
  >
    <EmptyStateIcon
      color="var(--pf-global--success-color--100)"
      icon={[Function]}
    />
    <Title
      headingLevel="h5"
      size="lg"
    >
      Configuration successful
    </Title>
    <EmptyStateBody>
      Here I Am
    </EmptyStateBody>
    <Button
      onClick={[MockFunction]}
      variant="primary"
    >
      Go back to my application
    </Button>
  </EmptyState>
</div>
`;

exports[`Steps components LoadingStep renders correctly 1`] = `
<div
  className="pf-l-bullseye"
>
  <EmptyState
    variant="full"
  >
    <div>
      <Spinner />
    </div>
    <EmptyStateBody>
      Loading, please wait.
    </EmptyStateBody>
  </EmptyState>
</div>
`;

exports[`Steps components LoadingStep renders correctly with custom props 1`] = `
<div
  className="pf-l-bullseye"
>
  <EmptyState
    variant="full"
  >
    <div>
      <Spinner />
    </div>
    <EmptyStateBody>
      Here I Am
    </EmptyStateBody>
    <EmptyStateSecondaryActions>
      <Button
        onClick={[MockFunction]}
        variant="link"
      >
        Cancel
      </Button>
    </EmptyStateSecondaryActions>
  </EmptyState>
</div>
`;
